iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesParser.java

```diff
@@ -47,21 +47,28 @@ public KubernetesParser(HelmProcessor helmProcessor) {
 
   @Override
   public FileTree parse(String source, @Nullable InputFileContext inputFileContext) {
-    if (hasHelmContent(source)) {
-      if (inputFileContext != null) {
-        var filename = inputFileContext.inputFile.filename();
-        LOG.debug("Helm content detected in file '{}'", inputFileContext.inputFile);
-        var evaluatedSource = helmProcessor.processHelmTemplate(filename, source, inputFileContext);
-        if (evaluatedSource != null) {
-          return super.parse(evaluatedSource, inputFileContext, FileTree.Template.HELM);
-        }
-      } else {
-        LOG.debug("No InputFileContext provided, skipping processing of Helm file");
-      }
+    if (!hasHelmContent(source)) {
+      return super.parse(source, inputFileContext);
+    } else {
+      return parseHelmFile(source, inputFileContext);
+    }
+  }
 
-      return super.parse("{}", inputFileContext, FileTree.Template.HELM);
+  private FileTree parseHelmFile(String source, @Nullable InputFileContext inputFileContext) {
+    if (inputFileContext == null) {
+      LOG.debug("No InputFileContext provided, skipping processing of Helm file");
+      return super.parse("{}", null, FileTree.Template.HELM);
```

Comment:
Not related to this ticket, but I'm really unsure why we are always null checking here. I checked the whole project, and this may be something we could improve overall... (But this should not be done here)

Reply:
Yes, I also thought about it. I found that we sometimes pass `null` there in Docker, but not in other extensions